### ISSUE TITLE
Barchart label not always correct

### ISSUE THREAD

**Issue by inodb** [cBioPortal/cbioportal#8375]
See number of bone scans in https://genie-private.cbioportal.org/study/summary?id=nsclc_genie_bpc

![Screen Shot 2021-02-25 at 9 17 58 AM](https://user-images.githubusercontent.com/1334004/109166345-9c63bf00-774a-11eb-84ac-c87085ee2d35.png)

The data is mostly 0 and sometimes positive but never negative. Either the labels should be 0 and >0, but in this case since there are not that many ticks, maybe it makes sense to show all ticks?

Same for:

![image](https://user-images.githubusercontent.com/1334004/109168823-32004e00-774d-11eb-9a42-a86cc799f9d7.png)

Prolly better to show all ticks

**Comment by onursumer** [comment]
I don't think it is straightforward to modify the existing logic to support mixing discrete values, like `0`, with range values, like `> 0`. Finding a generic solution for these kind of edge cases seems tricky. We need to figure out how to generate distinct bins for certain values, and range bins for the rest without breaking existing charts.

Trying to generate distinct bins for all the values might be an easier solution for now. Seems like we have hardcoded the threshold to `5`:

https://github.com/cBioPortal/cbioportal/blob/a53b5021e77a8d7131b4225032d971a445cf1491/web/src/main/java/org/cbioportal/web/util/DataBinner.java#L261

We can increase this value to `10` and hope that it solves the problem for most of the cases. We would still have the same issue if the data set contains 11 or more distinct values though.

Another solution could be to add custom logic for certain type of clinical attributes like we did for `age` attributes:

https://github.com/cBioPortal/cbioportal/blob/8ce576569c868118079889929f1c60251c65e657/web/src/main/java/org/cbioportal/web/util/LinearDataBinner.java#L51

Not sure how to categorize these type of attributes though.
